File: maven-root/mtsa/src/main/java/RobotProtocol/src/main/java/robotProtocol/messages/ReadIRSensorsMessage.cpp
/*
 * ReadIRSensorsMessage.cpp
 *
 *  Created on: Jun 24, 2013
 *      Author: mariano
 */

#include "ReadIRSensorsMessage.h"


namespace RobotProtocol {

ReadIRSensorsMessage ReadIRSensorsMessage::staticMessage = ReadIRSensorsMessage(0,0);

Message *ReadIRSensorsMessage::deserialize(char *serializedMessage){
	ReadIRSensorsMessage::staticMessage.initialize(serializedMessage);
	return &(ReadIRSensorsMessage::staticMessage);
}

ReadIRSensorsMessage::ReadIRSensorsMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

ReadIRSensorsMessage::ReadIRSensorsMessage(char *serializedMessage){
	initialize(serializedMessage);
}

ReadIRSensorsMessage::~ReadIRSensorsMessage() {
}

int ReadIRSensorsMessage::getMessageLength(){
	return Message::getMessageLength();
}

void ReadIRSensorsMessage::readData(char **serializedMessageIndex){}

void ReadIRSensorsMessage::writeData(char **serializedMessageIndex){}


} /* namespace RobotProtocol */

File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/ReadIRSensorsMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class ReadIRSensorsMessage extends Message {	
	public static int MESSAGE_ID				= 208;
	
	public ReadIRSensorsMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public ReadIRSensorsMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return true;
	}	
}
